opm/simulators/timestepping/AdaptiveTimeStepping_impl.hpp

```diff
@@ -720,7 +716,6 @@ runStepReservoirCouplingSlave_()
         report += sub_steps_report;
         current_time += timestep;
         if (final_step) {
-            substep_done = true;
```

Comment:
If you're removing this you might as well write the loop as
```C++
while (true) {
    // ...
}
```
since the condition `!substep_done` never changes.  Using `while (true)` also conditions the reader to look for a `break` statement.

Reply:
right you are. i only followed the 'unused assignment' warning, but didn't look close enough to realize it is completely unused otherwise. Have remedied, now in a separate commit. Also sprinkled some const.